### PR TITLE
Option for ignored columns on update

### PR DESCRIPTION
**Pull request by aqeelvn**
Closes https://github.com/jamis/bulk_insert/issues/50

Alternately I also noticed https://github.com/jamis/bulk_insert/pull/49 which also attempts to fix the same issue but the API interface is slightly different.
I could be wrong, but I'm guessing typically you may want to avoid only a small number of columns on `UPDATE` so rather than having to explicitly define all the columns it might be more convenient to just specify those columns that need to be ignored. 
Also this interface would nicely allow for having convenient default values like `created_at`. I haven't added it though in this PR as that could potentially be a breaking change and may need a major version bump depending on what the maintainers think.

cc @jamis @mberlanda

### CODE REVIEW

**Comment by aqeelvn** [comment]
The CI failure seems to be unrelated

File: lib/bulk_insert/worker.rb [hunk 1]
```diff
@@ -16,12 +16,14 @@ def initialize(connection, table_name, primary_key, column_names, set_size=500,
       @ignore = ignore
       @update_duplicates = update_duplicates
       @return_primary_keys = return_primary_keys
+      @ignored_columns_on_update = Array.wrap(ignored_columns_on_update)
```
Comment (mberlanda) [inline lib/bulk_insert/worker.rb]: @jamis I am not sure if the gem is supposed to work only with a dependency on `activerecord` or it was designed to have a dependency on `activesupport` as well:

https://apidock.com/rails/Array/wrap/class

Comment (aqeelvn) [reply lib/bulk_insert/worker.rb]: Ah, good spot. I don't think we need to keep this, and can use `Array(ignored_columns..)` as we are not expecting anything unusual in that param like a `hash`.

**Comment by aqeelvn** [comment]
Thank you for taking a look @mberlanda, appreciate it. I've applied your suggestions. Could you take another look when you get a chance?

**Comment by aqeelvn** [comment]
@mberlanda Thank you. Have ignored `ruby-head` for now. Would you like an issue to track this bit and handle it later ?

**Comment by mberlanda** [comment]
I would wait for @jamis feedback before merging.

I was considering to go for a major bump dropping the support for ruby < 2.4 and rails < 5.
This should be done definitely in a separated pr

**Comment by aqeelvn** [comment]
Hi @jamis , just wanted to quickly check if you had a chance to take a look at this one. Would appreciate any feedback you have.

**Comment by mberlanda** [comment]
@aqeelvn I hope you are doing great! Would you be available to re-submit your changes in a new revision of this PR rebasing on master?
I managed to make the test suite more resilient (testing against multiple ruby and rails version and different adapters). An adapter based design has been implemented for the query serialization.

Thank you very much for the time you spent on this contribution

**Comment by mberlanda** [comment]
hey @aqeelvn do you want me to cherry-pick your changes and adapt them to the new gem structure?

**Comment by aqeelvn** [comment]
@mberlanda Hey, thanks for reaching out. Yes it would be good to get the changes one way or the other. I know @sikachu were trying to get my changes on the latest master few days ago. @sikachu are you still working on it?

If not it would be great if you can cherry-pick my changes as you suggested, sorry I'm going through a busy schedule lately.

**Comment by sikachu** [comment]
@aqeelvn I'm also focusing on something else right now so @mberlanda please feel free to work on this.

**Pull request closed without merging**